docarray/array/array.py

```diff
@@ -386,18 +448,425 @@ def traverse_flat(
+                        compress=compress,
+                    )
+                    pbar.update(
+                        t, advance=1, total_size=str(filesize.decimal(_total_size))
+                    )
+
+    @classmethod
+    def load_binary(
+        cls: Type[T],
+        file: Union[str, bytes, pathlib.Path, io.BufferedReader, _LazyRequestReader],
+        protocol: str = 'pickle-array',
+        compress: Optional[str] = None,
+        _show_progress: bool = False,
+        streaming: bool = False,
+    ) -> Union[T, Generator['BaseDocument', None, None]]:
+        """Load array elements from a compressed binary file.
+
+        :param file: File or filename or serialized bytes where the data is stored.
+        :param protocol: protocol to use
+        :param compress: compress algorithm to use
+        :param _show_progress: show progress bar, only works when protocol is `pickle` or `protobuf`
+        :param streaming: if `True` returns a generator over `Document` objects.
+        In case protocol is pickle the `Documents` are streamed from disk to save memory usage
+        :return: a DocumentArray object
+
+        .. note::
+            If `file` is `str` it can specify `protocol` and `compress` as file extensions.
+            This functionality assumes `file=file_name.$protocol.$compress` where `$protocol` and `$compress` refer to a
+            string interpolation of the respective `protocol` and `compress` methods.
+            For example if `file=my_docarray.protobuf.lz4` then the binary data will be loaded assuming `protocol=protobuf`
+            and `compress=lz4`.
+        """
+        load_protocol: Optional[str] = protocol
+        load_compress: Optional[str] = compress
+        file_ctx: Union[nullcontext, io.BufferedReader]
+        if isinstance(file, (io.BufferedReader, _LazyRequestReader, bytes)):
+            file_ctx = nullcontext(file)
+        # by checking path existence we allow file to be of type Path, LocalPath, PurePath and str
+        elif isinstance(file, (str, pathlib.Path)) and os.path.exists(file):
+            load_protocol, load_compress = _protocol_and_compress_from_file_path(
+                file, protocol, compress
+            )
+            file_ctx = open(file, 'rb')
+        else:
+            raise FileNotFoundError(f'cannot find file {file}')
+        if streaming:
+            return cls._load_binary_stream(
+                file_ctx,
+                protocol=load_protocol,
+                compress=load_compress,
+                _show_progress=_show_progress,
+            )
+        else:
+            return cls._load_binary_all(
+                file_ctx, load_protocol, load_compress, _show_progress
+            )
+
+    def save_binary(
+        self,
+        file: Union[str, pathlib.Path],
+        protocol: str = 'pickle-array',
+        compress: Optional[str] = None,
+    ) -> None:
+        """Save DocumentArray into a binary file.
```

Comment:
We need more in the docstring IMO.

Here it is the place to explain the difference between the different serilization method etc. When to pick `pickle-array` over `protobuf-array`